new-cdc/eventuate-local-java-cdc-connector-mysql-binlog/src/main/java/io/eventuate/local/mysql/binlog/MySqlBinaryLogClient.java

```diff
@@ -230,21 +240,21 @@ private void handleWriteRowsEvent(Event event, Optional<BinlogFileOffset> starti
     offsetStore.save(binlogFileOffset);
   }
 
-//  private void handleUpdateRowsEvent(Event event) {
-//    UpdateRowsEventData eventData = event.getData();
-//
-//    if (eventData == null) {
-//      return;
-//    }
-//
-//    if (cdcMonitoringTableId.map(id -> id.equals(eventData.getTableId())).orElse(false)) {
-//      onMonitoringEventReceived();
-//    }
-//
-//    offset = ((EventHeaderV4) event.getHeader()).getPosition();
-//    BinlogFileOffset binlogFileOffset = new BinlogFileOffset(binlogFilename, offset);
-//    offsetStore.save(binlogFileOffset);
-//  }
+  private void handleUpdateRowsEvent(Event event) {
+    UpdateRowsEventData eventData = event.getData();
+
+    if (eventData == null) {
+      return;
+    }
+
+    if (cdcMonitoringTableId.map(id -> id.equals(eventData.getTableId())).orElse(false)) {
+      onMonitoringEventReceived(timestampExtractor.extract(MONITORING_SCHEMA_AND_TABLE, eventData));
+    }
+
+    offset = ((EventHeaderV4) event.getHeader()).getPosition();
```

Comment:
These 3 lines are important and occur more than once in this class. Extract into method

Reply:
ok